mmc-google-analytics/src/test/unit/java/info/novatec/metricscollector/google/RequestBuilderTest.java

```diff
@@ -0,0 +1,188 @@
+        requestBuilder.addMetrics(expectedMetrics);
+        assertThat(requestBuilder.getMetrics().size()).isEqualTo(3);
+        requestBuilder.getMetrics()
+            .forEach(actualMetric -> assertThat(expectedMetrics.contains(actualMetric.getExpression())));
+    }
+
+    @Test
+    public void add1Dimension() {
+        requestBuilder.addDimensions(Collections.singletonList(DataProvider.GA_DIMENSION));
+        assertThat(requestBuilder.getDimensions().size()).isEqualTo(1);
+        assertThat(requestBuilder.getDimensions().get(0).getName()).isEqualTo(DataProvider.GA_DIMENSION);
+    }
+
+    @Test
+    public void add3Dimensions() {
+        List<String> expectedDimensions = DataProvider.createGoogleAnalyticsDimensions(3);
+        requestBuilder.addDimensions(expectedDimensions);
+        assertThat(requestBuilder.getDimensions().size()).isEqualTo(3);
+        requestBuilder.getDimensions()
+            .forEach(actualDimension -> assertThat(expectedDimensions.contains(actualDimension.getName())));
+    }
+
+    @Test
+    public void addDimensionFilterWithNotOperator(){
+        requestBuilder
+            .addDimensionFilter(
+                DataProvider.GA_DIMENSION,
+                DimensionFilterOperators.NOT,
+                Collections.singletonList(DataProvider.DIMENSION_VALUE));
+        assertThat(requestBuilder.getDimensionFilters().get(0).getOperator()).isEqualTo(DimensionFilterOperators.EXACT.name());
+        assertThat(requestBuilder.getDimensionFilters().get(0).getNot()).isEqualTo(true);
+    }
+
+    @Test
+    public void addDimensionFilter() {
+        requestBuilder.addDimensionFilter(
+            DataProvider.GA_DIMENSION,
+            DimensionFilterOperators.EXACT,
+            Collections.singletonList(DataProvider.DIMENSION_VALUE));
+        assertThat(requestBuilder.getDimensionFilters().size()).isEqualTo(1);
+        assertThat(requestBuilder.getDimensionFilters().get(0).getDimensionName()).isEqualTo(DataProvider.GA_DIMENSION);
+    }
+
+    @Test
+    public void setDateRange() {
+        LocalDate endDate = LocalDate.now();
+        LocalDate startDate = endDate.minusDays(1);
+        requestBuilder.setDateRange(startDate, endDate);
+
+        assertThat(requestBuilder.getDateRange().getStartDate()).isEqualTo(startDate.toString());
+        assertThat(requestBuilder.getDateRange().getEndDate()).isEqualTo(endDate.toString());
+    }
+
+    @Test
+    public void createDimensionFilterClause() {
+        List<DimensionFilter> dimensionFilters =
+            Collections.singletonList(new DimensionFilter().setDimensionName(DataProvider.GA_DIMENSION));
+        DimensionFilterClause filterClause = requestBuilder.createDimensionFilterClause("AND", dimensionFilters);
+
+        assertThat(filterClause.getFilters().size()).isEqualTo(1);
+        assertThat(filterClause.getFilters().get(0).getDimensionName()).isEqualTo(DataProvider.GA_DIMENSION);
+    }
+
+    @Test
+    public void createReport(){
```

Comment:
Is here the createReportRequest() method tested, I don't see where it is called in test?

Reply:
Line 142. Maybe it confused you since the method name isn't that specific? It should be "createReportRequest()"